src/threadsync.cpp

```diff
@@ -461,14 +469,14 @@ bool ThreadSync::wrapperExecutionLockLockExcl()
 void ThreadSync::wrapperExecutionLockUnlock()
 {
   int saved_errno = errno;
-  if (WorkerState::currentState() != WorkerState::RUNNING &&
-      !DmtcpWorker::exitInProgress()) {
-    fprintf(stderr, "DMTCP INTERNAL ERROR: %s:%d: %s\n"
-            "       This process is not in RUNNING state and yet this thread\n"
-            "       managed to acquire the wrapperExecutionLock.\n"
-            "       This should not be happening, something is wrong.\n",
-            __FILE__, __LINE__, __PRETTY_FUNCTION__);
-    _exit(1);
+  // Ignore locks if we are about to exit
+  /* XXX:
+   *   One of the user threads could have called exit(), setting the
+   *   exitInProgress flag, and changing the WorkerState to Unknown. Does it
+   *   matter if we ignore the lock?
+   */
+  if (DmtcpWorker::exitInProgress()) {
```

Comment:
We already have a global per-thread global variable that counts The number of times this lock was acquired (_wrapperExecutionLockLockCount). If it's trivial to ise that variable to add the warning, let's do that, otherwise please add a Todo in there and then we can commit it.